### PR TITLE
GNOME Shell Sass rework

### PR DESCRIPTION
**Pull request by nana-4**
- Make `_common.scss` modular following upstream
- Use `$version` variable instead of having versioned SCSS files
- Drop support for versions 3.18 - 3.24
- Add support for version 3.36

### CODE REVIEW

**Comment by timocapa** [comment]
Not sure if you saw this, but there's an issue in the power menu as well.

Materia:
image

Default:
image

**Comment by nana-4** [comment]
@timocapa Thanks for testing. I'll fix that.

**Comment by nana-4** [comment]
For double check:

- [ ] _a11y.scss
- [ ] _app-grid.scss
- [ ] _base.scss
- [ ] _buttons.scss
- [ ] _calendar.scss
- [ ] _check-box.scss
- [ ] _corner-ripple.scss
- [ ] _dash.scss
- [ ] _dialogs.scss
- [ ] _entries.scss
- [ ] _hotplug.scss
- [ ] _ibus-popup.scss
- [ ] _keyboard.scss
- [ ] _login-dialog.scss
- [ ] _looking-glass.scss
- [ ] _message-list.scss
- [ ] _misc.scss
- [ ] _network-dialog.scss
- [ ] _notifications.scss
- [ ] _osd.scss
- [ ] _overview.scss
- [ ] _panel.scss
- [ ] _popovers.scss
- [ ] _screen-shield.scss
- [ ] _scrollbars.scss
- [ ] _search-entry.scss
- [ ] _search-results.scss
- [ ] _slider.scss
- [ ] _switcher-popup.scss
- [ ] _switches.scss
- [ ] _tiled-previews.scss
- [ ] _window-picker.scss
- [ ] _workspace-switcher.scss
- [ ] _workspace-thumbnails.scss

**Comment by peterychuang** [comment]
After re-generating the css files, I see a few of these text entries places where the text becomes dark in normal and light variants. For example, in the search entry of the gnome-shell overview:

Screenshot from 2020-03-12 08-56-09

This also happens on login screen and screen lock.

I've been poking around for a while and haven't been able to figure out why.

**Comment by peterychuang** [comment]
A quick patch to fix the dividers in menus.

```
---
 .../gnome-shell/components/menu/_menu.scss    | 20 ++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/src/_sass/gnome-shell/components/menu/_menu.scss b/src/_sass/gnome-shell/components/menu/_menu.scss
index 4eb0a330..e389f3ce 100644
--- a/src/_sass/gnome-shell/components/menu/_menu.scss
+++ b/src/_sass/gnome-shell/components/menu/_menu.scss
@@ -134,14 +134,20 @@
 
 // separator
 .popup-separator-menu-item {
-  // -margin-horizontal: 24px;
-  height: 1px; // not really the whole box
-  margin: 8px 64px - 24px;
-  background-color: $divider;
+  padding: 0;
 
-  border-color: transparent;
-  border-bottom-width: 0;
-  border-bottom-style: none;
+  .popup-separator-menu-item-separator {
+    //-margin-horizontal: 24px;
+    height: 1px; //not really the whole box
+    margin: 8px 64px - 24px;
+    background-color: $divider;
+    .popup-sub-menu & { //submenu separators
+      margin: 0 64px 0 32px;
+      @if $variant == 'dark' {
+        background-color: transparent;
+      }
+    }
+  }
 }
 
 // desktop background menu
-- 
2.25.1

```

File: src/_sass/gnome-shell/components/gdm/_screen-shield.scss [hunk 1]
```diff
@@ -11,73 +9,83 @@ $_screenshield_shadow: 0px 0px 6px rgba(0, 0, 0, 0.726);
...
   color: white;
-  text-shadow: $_screenshield_shadow;
-  font-weight: bold;
+  text-shadow: $shadow-1;
+  font-weight: normal;
   text-align: center;
   padding-bottom: 1.5em;
 }
 
 .screen-shield-clock-time {
```
Comment (peterychuang) [inline src/_sass/gnome-shell/components/gdm/_screen-shield.scss]: .screen-shield-* suffix appears to have been changed to .unlock-dialog-* in the upstream

https://gitlab.gnome.org/GNOME/gnome-shell-sass/-/blob/master/widgets/_screen-shield.scss

**Comment by nana-4** [comment]
> After re-generating the css files, I see a few of these text entries places where the text becomes dark in normal and light variants.

@peterychuang Thanks for testing. Fixed now :)

File: src/_sass/gnome-shell/components/gdm/_screen-shield.scss (hunk 1 above)
Comment (nana-4) [reply src/_sass/gnome-shell/components/gdm/_screen-shield.scss]: Fixed.

**Pull request merged**